packages/routing-utils/src/superstatic.ts

```diff
@@ -174,28 +182,60 @@ export function sourceToRegex(
   return { src: r.source, segments };
 }
 
+function collectHasSegments(has?: HasField) {
+  const hasSegments = new Set<string>();
+
+  for (const hasItem of has || []) {
+    if (!hasItem.value && (hasItem as any).key) {
```

Comment:
Can you avoid `as any` here by doing `key in hasItem`?

Reply:
Updated 👍